### ISSUE TITLE
Version file causes a fatal error in foreach()

### ISSUE THREAD

**Issue by hdodov** [distantnative/retour-for-kirby#154]
I installed Retour and got the following error:

![image](https://user-images.githubusercontent.com/5570098/75054458-c2bda180-54db-11ea-9cc0-de0b6256bfd0.png)

On the next request, the `.retour` file was created and I was no longer getting the error.

---

I think it's redundant to have just the version of the plugin in a separate file that could even cause errors. I can see that the reason for that is to know if the contained data has the correct schema according to the plugin version, but is that _really_ needed? How often would you change that schema? How severe consequences will that have? I think it's a bit over-engineered.

At the very least, you could store that version in the `redirects.yml` and have:

```yml
schema: "2.3.1"
entries:
  - 
    status: "200"
    from: foo
    to: /bar
    hits: "0"
    last: null
  - 
    status: "200"
    from: bar
    to: baz
    hits: "0"
    last: null
```

I find it kind of annoying that I have to track a file in Git that contains just some version number. Why is that file there? What is it used for? It's polluting.

---

All of that aside - the plugin rocks! 🤘

**Comment by distantnative** [comment]
You are right. Actually, I will get rid of the whole update code for the next version. Schema changes are so rarely that they don't make up the bugs etc. from the update code.

If there are schema changes in the future, they will happen between major versions and I will provide update instructions/script.

✅

**Comment by adamkiss** [comment]
Actually, the problem still persists, and I just had to manually create the retour file.

The problem is, AFAIK, here: https://github.com/distantnative/retour-for-kirby/blob/81e244e979789510fcb46de3a4e809ead9cfdcb7/src/models/Version.php#L65-L66

The `usort` [returns true or false](https://www.php.net/manual/en/function.usort.php#refsect1-function.usort-returnvalues) based on the success of the sorting, and modifies the original array. I don't see how this code ever worked - I don't think it ever did, but I don't see how this plugin might be in the wild for months with this sort of issue.

**Comment by distantnative** [comment]
I’ll be very honest, I thought I could be out with a major new version already. I banked on the new UI components we were working on for the core... but now those had to be pushed back and with that also Retour 3, which is a big rewrite that handles things very differently internally.

But you are totally right, that it’s a bad state where a broken version is out for such a long time without a fix.

Will try to find a fix for Retour 2 on the weekend, so it still works until Retour 3 comes out.

**Comment by hdodov** [comment]
As a side note, I think the schema version doesn't have to follow the version of the plugin itself. I mean, you could have schema version 1, 2, 3... just simple incremental numbers. It doesn't make sense to change the schema version on each plugin release when there's no actual change in the schema itself.

**Comment by distantnative** [comment]
I published 2.3.2 that should fix your problems: https://github.com/distantnative/retour-for-kirby/releases/tag/2.3.2

@hdodov You are right, it doesn't have to. But I thought it's easier than keeping track of separate schema version numbers. And finding ways to provide the info which plugin version number corresponds to which schema number. Also, I called it schema... but it's not really just about the config file or database schema, but could also apply to where those files are located or other things.
